app/models/agents/local_file_agent.rb

```diff
@@ -0,0 +1,158 @@
+module Agents
+  class LocalFileAgent < Agent
```

Comment:
This should probably check `ENABLE_INSECURE_AGENTS` like the `ShellCommandAgent`?